src/Analyser/TypeSpecifier.php

```diff
@@ -234,6 +235,13 @@ public function specifyTypesInCondition(
 						}
 					}
 				}
+
+				if ($context->falsey() && $constantType instanceof ConstantStringType) {
```

Comment:
I did not understand the change fully, but would this also make sense for other constant types here?

Reply:
Tbh I'm also not a 100% sure if this is the right/best fix. But I thought the same, yeah. Will take a look later. Also thought of getting an opinion from Ondrej first maybe